packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -686,17 +689,26 @@ class DropdownMenuItem<T> extends _DropdownMenuItemContainer {
     Key? key,
     this.onTap,
     this.value,
+    this.disabled = false,
     required Widget child,
-  }) : assert(child != null),
-       super(key: key, child: child);
+  })   : assert(child != null),
+        super(key: key, child: child);
 
-  /// Called when the dropdown menu item is tapped.
+  /// Called when the dropdown menu item is tapped, provided [disabled] is false.
   final VoidCallback? onTap;
 
-  /// The value to return if the user selects this menu item.
+  /// The value to return if the user selects this menu item, provided [disabled] is false.
   ///
   /// Eventually returned in a call to [DropdownButton.onChanged].
   final T? value;
+
+  /// If `true`:
+  /// * [value] and [onTap] won't be used.
+  /// * the item will not be clickable / focusable.
+  /// * tapping won't close drop down list.
+  ///
+  /// defaults to `false`
```

Comment:
We usually have the following structure for API docs:

/// A single line to succinctly describe the parameter or API.
///
/// More content to further elaborate on the API itself. This can span multiple
/// lines to explain the purpose of the API and how to use it in a more elaborate
/// way.
///
/// Sometimes, describing what the default behavior is. (ie. If `null`, then [parameter] is used).
///
/// Sometimes, a `See more:` section (see other API docs for an idea of how this looks like)

Reply:
Is this okay?
```dart
/// Whether a user can select this menu item. Defaults to `false`.
///
/// If `true`: [value] and [onTap] should be `null`
/// and user will **not** be able to select this item.
///
/// if `false`: user will be able to select this item.
```
<img src=https://user-images.githubusercontent.com/44755140/110444886-8b6c6380-80e3-11eb-86a9-860738c107d2.png width=500>
Now I feel the name of the variable should be [enabled] or [isEnabled] instead of [disabled]. What do you say?

Reply:
`enabled` or `isEnabled` is fine too.  Remember, the first sentence should only be one line. Here are some suggestions.
```
/// Whether or not a user can select this menu item. 
/// 
/// If `true`, [value] and [onTap] should be `null` and the user 
/// will **not** be able to select this item. If `false`, then the user 
/// will be able to select this item.
/// 
/// Defaults to `false`.
```

Of course, change the wording based on whether you flip the property name to `enabled` or not.

Reply:
You may also want to consider explaining why `value` and `onTap` should be `null` here.

Reply:
Is this okay?
```dart
/// Whether or not a user can select this menu item.
///
/// If `false`, the user will be unable to select this item,
/// and since [value] and [onTap] are redundant for a disabled
/// and unselectable menu item, both should be `null`.
/// If `true`, the user will be able to select this item.
///
/// Defaults to `true`
```